File: WCPMVCCS/Controllers/DemoPrintFilePDFController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

using Neodynamic.SDK.Web;

namespace WCPMVCCS.Controllers
{
    public class DemoPrintFilePDFController : Controller
    {
        // GET: DemoPrintFile
        public ActionResult Index()
        {
            ViewBag.WCPScript = Neodynamic.SDK.Web.WebClientPrint.CreateScript(Url.Action("ProcessRequest", "WebClientPrintAPI", null, HttpContext.Request.Url.Scheme), Url.Action("PrintFile", "DemoPrintFilePDF", null, HttpContext.Request.Url.Scheme), HttpContext.Session.SessionID);

            return View();
        }

        [AllowAnonymous]
        public void PrintFile(string printerName, string trayName, string paperName, string printRotation, string pagesRange, string printAnnotations, string printAsGrayscale, string printInReverseOrder, string manualDuplexPrinting, string driverDuplexPrinting, string pageSizing, bool autoRotate, bool autoCenter)
        {
            if (manualDuplexPrinting == "true" && driverDuplexPrinting == "true")
            {
                manualDuplexPrinting = "false";
            }

            string fileName = Guid.NewGuid().ToString("N");
            string filePath = filePath = "~/files/mixed-page-orientation.pdf";

            PrintFilePDF file = new PrintFilePDF(System.Web.HttpContext.Current.Server.MapPath(filePath), fileName);
            file.PrintRotation = (PrintRotation)Enum.Parse(typeof(PrintRotation), printRotation); ;
            file.PagesRange = pagesRange;
            file.PrintAnnotations = (printAnnotations == "true");
            file.PrintAsGrayscale = (printAsGrayscale == "true");
            file.PrintInReverseOrder = (printInReverseOrder == "true");
            if (manualDuplexPrinting == "true")
            {
                file.DuplexPrinting = true;
                //file.DuplexPrintingDialogMessage = "Your custom dialog message for duplex printing";
            }
            file.Sizing = (Sizing)Enum.Parse(typeof(Sizing), pageSizing);
            file.AutoCenter = autoCenter;
            file.AutoRotate = autoRotate;

            ClientPrintJob cpj = new ClientPrintJob();
            cpj.PrintFile = file;
            if (printerName == "null")
                cpj.ClientPrinter = new DefaultPrinter();
            else
            {
                if (trayName == "null") trayName = "";
                if (paperName == "null") paperName = "";

                cpj.ClientPrinter = new InstalledPrinter(printerName, true, trayName, paperName);

                if (driverDuplexPrinting == "true")
                    ((InstalledPrinter)cpj.ClientPrinter).Duplex = Duplex.Vertical;
            }

           
            System.Web.HttpContext.Current.Response.ContentType = "application/octet-stream";
            System.Web.HttpContext.Current.Response.BinaryWrite(cpj.GetContent());
            System.Web.HttpContext.Current.Response.End();
            
        }
    }
}

File: WCPMVCVB/Controllers/DemoPrintFilePDFController.vb
Imports System.Web.Mvc

Imports Neodynamic.SDK.Web


Namespace Controllers
    Public Class DemoPrintFilePDFController
        Inherits Controller

        ' GET: DemoPrintFilePDF
        Function Index() As ActionResult

            ViewData("WCPScript") = Neodynamic.SDK.Web.WebClientPrint.CreateScript(Url.Action("ProcessRequest", "WebClientPrintAPI", Nothing, HttpContext.Request.Url.Scheme), Url.Action("PrintFile", "DemoPrintFilePDF", Nothing, HttpContext.Request.Url.Scheme), HttpContext.Session.SessionID)

            Return View()
        End Function


        <AllowAnonymous> Public Sub PrintFile(printerName As String, trayName As String, paperName As String, printRotation As String, pagesRange As String, printAnnotations As String, printAsGrayscale As String, printInReverseOrder As String, manualDuplexPrinting As String, driverDuplexPrinting As String, pageSizing As String, autoRotate As Boolean, autoCenter As Boolean)

            If (manualDuplexPrinting = "true" AndAlso driverDuplexPrinting = "true") Then
                manualDuplexPrinting = "false"
            End If

            Dim fileName As String = Guid.NewGuid().ToString("N")
            Dim filePath As String = "~/files/mixed-page-orientation.pdf"

            Dim File As New PrintFilePDF(System.Web.HttpContext.Current.Server.MapPath(filePath), fileName)
            File.PrintRotation = [Enum].Parse(GetType(PrintRotation), printRotation)
            File.PagesRange = pagesRange
            File.PrintAnnotations = (printAnnotations = "true")
            File.PrintAsGrayscale = (printAsGrayscale = "true")
            File.PrintInReverseOrder = (printInReverseOrder = "true")
            If (manualDuplexPrinting = "true") Then
                File.DuplexPrinting = True
                File.DuplexPrintingDialogMessage = "Your custom dialog message for duplex printing"
            End If
            File.Sizing = [Enum].Parse(GetType(Sizing), pageSizing)
            File.AutoCenter = autoCenter
            File.AutoRotate = autoRotate

            Dim cpj As New ClientPrintJob()
            cpj.PrintFile = File
            If (printerName = "null") Then
                cpj.ClientPrinter = New DefaultPrinter()
            Else
                If (trayName = "null") Then trayName = ""
                If (paperName = "null") Then paperName = ""

                cpj.ClientPrinter = New InstalledPrinter(printerName, True, trayName, paperName)

                If (driverDuplexPrinting = "true") Then
                    DirectCast(cpj.ClientPrinter, InstalledPrinter).Duplex = Duplex.Vertical
                End If

            End If

            System.Web.HttpContext.Current.Response.ContentType = "application/octet-stream"
            System.Web.HttpContext.Current.Response.BinaryWrite(cpj.GetContent())
            System.Web.HttpContext.Current.Response.End()

        End Sub

    End Class
End Namespace